pkg/logging/limit.go

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package logging
+
+import (
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"golang.org/x/time/rate"
+)
+
+type rateLimitedLogger struct {
+	logger     log.Logger
+	limiter    *rate.Limiter
+	occurrence int
+}
+
+func (r *rateLimitedLogger) Log(keyvals ...interface{}) error {
```

Comment:
We probably need to be bit smarter here.

* I would try to limit the same messages. Doing `level.Warn(logging.With(logger, limit1)).Log` does the work well so all good
* I would somehow don't lose the info about lost messages as we rely on them. Either some metrics for dropped logs or better: Another field to this message saying `occured` to mention how many of those occurred during last log time.

Reply:
How about now?